### PR TITLE
Cloud layer improvements

### PR DESCRIPTION
**Pull request by adrien-de-tocqueville**
### Purpose of this PR

Based on #6828 but a bit smaller. Most notably, this version doesn't sample the ambient probe.

**Changes:**

- Sun light now always tint cloud color (previoulsy _lighting_ option was sun color + raymarching, now sun color is always on and option is renamed to _raymarching_)
- added _altitude_ option for sun atmospheric attenuation
- renamed _thickness_ to density
- renamed _distortion_ to _wind_ (more consistent with volumetric clouds)

Following artist request, minor changes to the visual environment:
Default value for wind is now 0
Info box when wind value is non zero and always refresh isn't enabled
image

Screenshot in what's new doc:

cl-whats-new

**Lighting Screenshots:**

Open in new tab to view them better. All screenshots use the same number of steps

|   | High Sun | Low Sun |
| ------------- | ------------- | ------------- |
| Branch | image | image
| HW | image | image
| Branch 2 (see note below) | image | image

NOTE 1: "Branch 2" is only the first commit on this branch, it doesn't include changes to the raymarching code and therefore it's what you would get on master but including the sun color improvements of this branch.
NOTE 2: i think the HW version looks slightly better than this branch but i'll start with that. Ambient probe makes the biggest diff, but it requires a bit more work to include it properly.

---
### Testing status

See screenshots above
Tested with artist

Yamato seems green, DRS failure is unrelated

Updated graphic test
5010_CloudLayer

### CODE REVIEW

**Review by pierre-unity — CHANGES_REQUESTED** [review]
With raymarching:
image

Without raymarching:
image


It's working well so far in my testing. The consistency with the sun light and the volumetric clouds is great. The removal of the "Exposure" control requires the user to play with the "Opacity" instead when NOT using the raymarching. This is somewhat acceptable, as the results are very similar. Up for discussion, if we want to keep it or not.

One problem: we need to indent the checkboxes for the Wind orientation and speed:
image

**Review by iM0ve — CHANGES_REQUESTED** [review]
**Whats tested:**
* Upgrading project
* Using the new properties
* Changing sunlight temperature and filter
* PB sky and HDRI sky
* Mac

**Concerns:**
1) There is a notable improvement when facing the suns direction (first two images), but when looking at the opposite direction clouds now always look gray/rainy and flat. Visually it looks very different to pre-upgrade and not many properties to alter their appearance. Not sure if this is an expected side effect

Sun facing (before)
Old2

Sun facing (after)
New2

Here is the concern, when camera is not facing the sun.

Before
Old NonBaked

After
New

2) Since exposure option is now removed. If lets say for performance reasons Raymarching cant be used for a project, the clouds get super bright and no way to reduce their intensity anymore.

exposure

**Review by iM0ve — APPROVED** [review]
I updated my review. All of the concerns were addressed. Works great, here is a showcase.

CloudLayer

File: com.unity.render-pipelines.high-definition/Runtime/Sky/CloudSystem/CloudLayer/CloudLayerRenderer.cs [hunk 1]
```diff
@@ -95,24 +95,41 @@ public override void RenderClouds(BuiltinSkyParameters builtinParams, bool rende
...
+
+            s_VectorArray[0] = cloudLayer.layerA.Color * lightColor; s_VectorArray[1] = cloudLayer.layerB.Color * lightColor;
+            s_VectorArray[0].w = cloudLayer.layerA.altitude.value; s_VectorArray[1].w = cloudLayer.layerB.altitude.value;
+            m_CloudLayerMaterial.SetVectorArray(HDShaderIDs._Params1, s_VectorArray);
+
+            // Keywords
             CoreUtils.SetKeyword(m_CloudLayerMaterial, "USE_CLOUD_MOTION", cloudLayer.layerA.distortionMode.value != CloudDistortionMode.None);
             if (cloudLayer.layerA.distortionMode.value != CloudDistortionMode.None)
-                cloudLayer.layerA.scrollFactor += cloudLayer.layerA.scrollSpeed.GetValue(hdCamera) * dt * 0.01f;
+                cloudLayer.layerA.scrollFactor += cloudLayer.layerA.scrollSpeed.GetValue(hdCamera) * dt * 0.277778f;
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Sky/CloudSystem/CloudLayer/CloudLayerRenderer.cs:132]: could you comment where this magic number is coming from ? .277778f;

**Review by sebastienlagarde — APPROVED** [review]
just a comment to add on the magic number

**Pull request merged**